### ISSUE TITLE
Block code with specific code kills any channel - urgent blocker and DOS attack

### ISSUE THREAD

**Issue by danpospisil** [RocketChat/Rocket.Chat#8959]
When you paste following code into any channel, it will completely freeze it in Chrome. Firefox takes a lot of time to recover and you don't even see the message. After somebody puts this code into a channel nobody is able to open it anymore without a lot of waiting or maybe never.

EDITED 

Please fix this ASAP. It looks that RC doesn't like the two (and more) dots. I tried it in my new channel even on https://open.rocket.chat and it froze also (EDITED) and its frozen now so you can see it for yourself...

**Comment by theblazehen** [comment]
@danpospisil Rather contact security@rocket.chat as per https://rocket.chat/docs/contributing/security/ and delete this issue

**Comment by mciverza** [comment]
This may be only relevant if using the original/homemade markdown parser.  As on our server using the the "real" markdown parser, we cannot replicate this.

**Comment by danpospisil** [comment]
@theblazehen - well, we have posted another issue there almost a year ago - no answer till now, so I don't think it's a good idea to send an email to some black hole when we need the fix ASAP (we have several locked channels).

**Comment by danpospisil** [comment]
@theblazehen - my colleague found one security issue once, posted it to their security email and after they wanted some registration and a lot of other stuff. And they haven't replied to that email ever. So since this issue is really important to us (we have several locked channels) I don't want it to end up in some black hole. I did go through the registration and sent it there - so if there is any reply from that I will delete the issue here.

**Comment by rodrigok** [comment]
Thanks guys, I'll take a look ASAP to fix this issue.

**Comment by danpospisil** [comment]
@rodrigok - thanks! So .. how can I delete this issue? I don't see delete button anywhere :/

**Comment by gdelavald** [comment]
@danpospisil I edited the message to remove the critical points. Thanks for you report.

**Comment by Rusty-1** [comment]
This seems to only effect the clients. I was still able to retrieve all messages quite successfully through the rest api. 
```   
 {
      "_id": "xxxxxxNfNJYAxBCidz",
      "rid": "xxxxxxxnMiA5J6dheHpPP5L9Jit9wLy5r",
      "msg": "```f(..)```",
      "ts": "2017-11-28T13:42:17.772Z",
      "u": {
        "_id": "eHpPP5L9Jit9wLy5r",
        "username": "rusty",
        "name": "Shaun Rust"
      },
      "mentions": [],
      "channels": [],
      "_updatedAt": "2017-11-28T13:42:17.774Z"
 },
```
Once I was able to identify the offending message id I also deleted it this way using.
```
curl -s -XPOST \
     -H "X-Auth-Token: ${TOKEN}" \
     -H "X-User-Id: ${USER_ID}" \
     -H "Accepts: application/json" \
     -H "Content-Type: application/json" \
     https://myrocketchat.com/api/v1/chat.delete \
     -d '{"roomId": "xxxxxxxnMiA5J6dheHpPP5L9Jit9wLy5r", "msgId":"xxxxxxNfNJYAxBCidz"}'
{
  "_id": "xxxxxxNfNJYAxBCidz",
  "ts": 1511886304454,
  "success": true
}


```

**Comment by rodrigok** [comment]
Guys, the issue was already solved in our develop version that will be available soon, I wasn't able to determine which PR/Change did the fix so I'll close this issue and assign it to version 0.60.

I'll try to fix it and release a minor version to 0.59 too.

Thank you by the report
